tests/testsuite/sbom/mod.rs

```diff
@@ -0,0 +1,336 @@
+//! Tests for cargo-sbom precursor files.
+
+use std::path::PathBuf;
+
+use cargo_test_support::basic_bin_manifest;
+use cargo_test_support::cargo_test;
+use cargo_test_support::compare::assert_e2e;
+use cargo_test_support::project;
+use cargo_test_support::registry::Package;
+use cargo_test_support::ProjectBuilder;
+
+const SBOM_FILE_EXTENSION: &str = ".cargo-sbom.json";
+
+fn with_sbom_suffix(link: &PathBuf) -> PathBuf {
+    let mut link_buf = link.clone().into_os_string();
+    link_buf.push(SBOM_FILE_EXTENSION);
+    PathBuf::from(link_buf)
+}
+
+fn configured_project() -> ProjectBuilder {
+    project().file(
+        ".cargo/config.toml",
+        r#"
+            [build]
+            sbom = true
+        "#,
+    )
+}
```

Comment:
Personally, I find helpers like this not too helpful
- They obfuscate what the test is doing (it should at least say what the config is)
- They don't scale if you then want to do something else with the config